File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/games/hitori/android/HitoriMainActivity.java
package com.zwstudio.logicpuzzlesandroid.games.hitori.android;

import com.zwstudio.logicpuzzlesandroid.R;
import com.zwstudio.logicpuzzlesandroid.common.android.MainActivity;
import com.zwstudio.logicpuzzlesandroid.games.hitori.data.HitoriDocument;
import com.zwstudio.logicpuzzlesandroid.games.hitori.domain.HitoriGame;
import com.zwstudio.logicpuzzlesandroid.games.hitori.domain.HitoriGameMove;
import com.zwstudio.logicpuzzlesandroid.games.hitori.domain.HitoriGameState;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.Click;
import org.androidannotations.annotations.EActivity;

@EActivity(R.layout.activity_hitori_main)
public class HitoriMainActivity extends MainActivity<HitoriGame, HitoriDocument, HitoriGameMove, HitoriGameState> {
    public HitoriDocument doc() {return app.hitoriDocument;}

    @AfterViews
    void init() {
        int[] levels = {1, 2, 3, 4, 5, 6, 7, 8, 16, 24, 34, 81};
        super.init(levels);
    }

    @Click
    void btnOptions() {
        HitoriOptionsActivity_.intent(this).start();
    }

    protected void resumeGame() {
        doc().resumeGame();
        HitoriGameActivity_.intent(this).start();
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/hitori/HitoriMainActivity.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.hitori

import android.content.Intent
import android.os.Bundle
import com.zwstudio.logicpuzzlesandroid.common.android.GameHelpActivity
import com.zwstudio.logicpuzzlesandroid.common.android.GameMainActivity
import com.zwstudio.logicpuzzlesandroid.common.android.GameOptionsActivity
import org.koin.android.ext.android.inject

class HitoriMainActivity : GameMainActivity<HitoriGame, HitoriDocument, HitoriGameMove, HitoriGameState>() {
    private val document: HitoriDocument by inject()
    override val doc get() = document

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding.btnOptions.setOnClickListener {
            startActivity(Intent(this, HitoriOptionsActivity::class.java))
        }
    }

    override fun resumeGame() {
        doc.resumeGame()
        startActivity(Intent(this, HitoriGameActivity::class.java))
    }
}

class HitoriOptionsActivity : GameOptionsActivity<HitoriGame, HitoriDocument, HitoriGameMove, HitoriGameState>() {
    private val document: HitoriDocument by inject()
    override val doc get() = document
}

class HitoriHelpActivity : GameHelpActivity<HitoriGame, HitoriDocument, HitoriGameMove, HitoriGameState>() {
    private val document: HitoriDocument by inject()
    override val doc get() = document
}
